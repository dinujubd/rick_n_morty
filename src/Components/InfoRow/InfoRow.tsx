import React from "react";
import { Grid, Typography } from "@material-ui/core"
import { useStyles } from "./InforRow.styles"

interface Props {
    name: string;
    value: string | number | React.ReactNode;
}

export const InfoRow: React.FC<Props> = React.memo(({ name, value }) => {
    const classes = useStyles();

    if (!value) return null;

    return (
        <Grid container justify="space-between">
            <Typography className={classes.dataTitle} variant="body1" gutterBottom>
                {name}
            </Typography>
            <Typography variant="body1" gutterBottom>
                {value}
            </Typography>
        </Grid>
    )
});

InfoRow.displayName = 'InfoRow';