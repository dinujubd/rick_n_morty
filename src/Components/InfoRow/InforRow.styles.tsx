import { createStyles, makeStyles, Theme } from "@material-ui/core";
import { PaletteColor } from "@material-ui/core/styles/createPalette";

export const useStyles = makeStyles(() =>
    createStyles({
        dataTitle: {
            fontWeight: 'bold'
        },
    }),
);