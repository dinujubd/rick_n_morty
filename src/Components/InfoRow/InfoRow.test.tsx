import renderer from 'react-test-renderer';
import { InfoRow } from './InfoRow';
describe('<InfoRow />', () => {
    it('should renders correctly', () => {
        const tree = renderer
            .create(<InfoRow name="Hi" value="Hello" />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});