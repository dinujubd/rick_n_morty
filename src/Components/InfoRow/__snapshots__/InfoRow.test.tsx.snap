// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InfoRow /> should renders correctly 1`] = `
<div
  className="MuiBox-root MuiBox-root-2"
>
  <div
    className="MuiBox-root MuiBox-root-3 makeStyles-dataTitle-1"
  >
    Hi
  </div>
  <div
    className="MuiBox-root MuiBox-root-4"
  >
    Hello
  </div>
</div>
`;
